utils/create_dataset.py

```diff
@@ -104,3 +114,155 @@ def __getitem__(self, index):
                                 f"\nIgnore if some augmentations have padded with dontcare value.")
         sample['index'] = index
         return sample
+
+
+class VRTDataset(GeoDataset):
```

Comment:
Why call it "**VRT**" Dataset? In practice, our AOI instances don't always create a VRT and will still use this class, right? This class can be used with any DatasetReader instance, be it VRT or regular dataset directly opened from a .tif file (ex.: multiband raster with no band selection or ordering), right ?

I'd suggest using the [check_rasterio_im_load](https://github.com/NRCan/geo-deep-learning/blob/develop/utils/geoutils.py#L147) utility to make this class a bit more generic. See comment below.

Reply:
Renamed it to the DRDataset (or, DatasetReader Dataset) because it deals directly with the DatasetReader object.
I did not use [check_rasterio_im_load](https://github.com/NRCan/geo-deep-learning/blob/develop/utils/geoutils.py#L147) because in lines 298-299 I open the raster with rasterion.open() if it was not opened before.